### DISCUSSION TITLE
Bulk version of the Catalog API

### DISCUSSION THREAD

**Discussion by DanielaZamoraUribe** [amzn/selling-partner-api-models#602] [Ideas]
It would be super useful to have an option to retrieve the catalog data for multiple ASINs all at once, so we don't have to loop over all our ASINs. 

My company uses the ASIN metadata to match it with our internal product metadata, but since we have noticed this data is not available in the Sales Diagnostic report generated by the API, the catalog API would be our best option.

**Comment by chapmanjw** [comment]
@DanielaZamoraUribe 

We are certainly looking for feedback on what mechanisms would be most useful for these use-cases. For your specific use-case, are you more looking for a report containing the same data as the Catalog Items API, but for all ASINs the selling partner has listings associated with? Or are you looking for an API mechanism that allows for more than one ASIN to be retrieved at a time (i.e., 10 or 20)?

**Reply by chapmanjw** [reply]
While we _might_ eventually do both, which would be your preference?

- ❤️ A `JSON_CATALOG_REPORT` for the Reports API containing ASIN data for all listings for the selling partner.
- 🚀 Extending the Catalog Items Search API to accept a list of identifiers (like `&identifiers={id-1},{id-2},{id-3}`) up-to 10 or 20.

**Reply by coopernate** [reply]
While both would definitely useful, my vote would be for the Report API containing all ASIN listings data for a given selling partner.

**Reply by michael-land** [reply]
The report usually update hourly. The dedicate api usually return data in real time.

**Reply by DanielaZamoraUribe** [reply]
My vote would be also for the Reports API containing all the ASINs. That would give us much more freedom to filter the results on the ASINs we are interested in, or have information for all of them in case it is needed.

**Reply by uk-2** [reply]
What we want is

- Able to pull listing down with product metadata (barcode,bullet points,description,product attributes ) 
-  Pull down & update brand owned metadata (barcode,bullet points,description,product attributes )

**Reply by Bowserlord** [reply]
I prefere having the ASIN before doing the first product-upload, i guess a search API with identifieres (e.g. EANS) would be great.

**Comment by NA-Dev** [comment]
I second this. I would love to be able to paginate over all our catalog ASINs. This way, we can easily capture when new ASINs are added to the catalog.

**Reply by sungolivia** [reply]
hey @NA-Dev  check this blog out that talks about bulk capabilities with Catalog Items API!
https://developer-docs.amazon.com/sp-api-blog/docs/reducing-api-calls-with-batch-functionality

**Reply by NA-Dev** [reply]
@sungolivia Thank you for the info, but this doesn't solve the entire problem. I would still love a way to just page over all our catalog without having to list 20 ASINs at a time. When new ASINs are added, I may nott have that new ASIN available in a list anywhere.
